staging/src/k8s.io/apiserver/pkg/storage/etcd3/store.go

```diff
@@ -826,7 +830,7 @@ func growSlice(v reflect.Value, maxCapacity int, sizes ...int) {
 	if len(sizes) == 0 || max > maxCapacity {
 		max = maxCapacity
 	}
-	if max <= cap {
+	if max <= cap && !v.IsNil() {
```

Comment:
maybe this is only confusing to me, but what are the implications for the following code if v _is_ nil?

if v is nil, cap will be 0, and if we end up wanting a max of 0, why is wrong to continue returning here with v still being nil?

Reply:
If the slice is nil and `cap` and `max` both equal 0, making `growSlice` turn a nil slice into a zero-length slice seems more confusing than helpful, at least to me.

Enforcing an invariant like "list will always return a non-nil result" in the higher level `GetList` function seems better.

Reply:
> Enforcing an invariant like "list will always return a non-nil result" in the higher level GetList function seems better.

OK - that makes sense. Fixed.